cmd/go-vcs/go-vcs.go

```diff
@@ -309,23 +310,38 @@ func main() {
 		}
 
 	case "branches":
-		if len(args) != 0 {
-			log.Fatal("branches takes no arguments.")
+		if len(args) > 1 {
+			log.Fatal("branches takes 0 or 1 arguments.")
+		}
+
+		// Open using go/vcs to figure out VCS type (git, hg).
+		r := vcs2.New(".")
```

Comment:
> Yeah, extracting the package into its own repo would be great.

I've created shurcooL/go#18 and I will look into addressing it eventually.

> Not a big deal either way, since this is in the command, not the lib (and Godeps users of this lib wouldn't pull in the cmd deps).

Agreed. The `cmd/go-vcs` is mainly used for testing/development, and the [installation instructions](https://github.com/sourcegraph/go-vcs#installing) specify:

```
go get sourcegraph.com/sourcegraph/go-vcs/vcs
```

Which means the dependencies of `go get sourcegraph.com/sourcegraph/go-vcs/cmd/go-vcs` will not be fetched, so it being a bigger repository will not affect those users.

That's why I'll leave it as is for now. It can be improved when doing next set of improvements/feature development for `go-vcs/vsc`.